### ISSUE TITLE
Why does tf.feature_column.input_layer return my feature values in rearranged order

### ISSUE THREAD

**Issue by jingwb222** [tensorflow/tensorflow#21751]
I currently am testing some ideas with `LinearClassifier` with `TensorFlow`'s New API. Have created some features using just `tf.feature_column.numeric_column` and feed them into a `tf.feature_column.input_layer` layer as following:

    layer_test = tf.feature_column.input_layer(input_fn('data.csv')[0], feature_columns_raw)

And then simply check if my input features are corrently fed via:

    with tf.Session() as sess:
        init = tf.global_variables_initializer()
        sess.run(init)
        layer_test_val = sess.run(layer_test)
        print(layer_test_val)

But I found if very strange and counterintuitive that column order in `layer_test_val` were completely changed out of my control. It's no big deal if the `LinearClassifier` can eventually get the decent prediction accuracy despite the rearranged features' order. However, if what I need is the learned weights of those features, how can I make sure that the output of 
`weights = lr_classifier.get_variable_value('logits/kernel')` is in the exact order of the features as they are specified.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by rohan100jain** [comment]
We sort the feature columns in order to ensure repeatibility and consistency. You can get the variable names from lr_classifier.get_variable_names(). The names of the weights have the feature names in them and you can use them to retrieve the values.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by facaiy** [comment]
@jingwb222 Hi, does @rohan100jain 's suggestion work for you?

**Comment by dksb** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by PiotrDabkowski** [comment]
If your code depends on feature ordering and you change the feature name it will just break. Why cant you simply ensure that the feature columns is a list/tuple and mention in the documentation that the order is the same as in the provided list? Also the current documentation does not mention the 'magic' ordering at all.

**Comment by wannibar** [comment]
mark. tf.feature_column.input_layer value is not in param order.

**Comment by JJplane** [comment]
> We sort the feature columns in order to ensure repeatibility and consistency. You can get the variable names from lr_classifier.get_variable_names(). The names of the weights have the feature names in them and you can use them to retrieve the values.

what does repeatibility and consistency mean?this is not convenient，because i can fetch the specific position column value in columns according to the input position if not sort. @rohan100jain
